magento/magento2
# Allow keyboard navigation in browser on product detail pages, by fixi…

…ng fotorama library to not change to a different image when pressing left or right arrow key in combination with certain modifier keys on your keyboard.

### Description
I'm trying to use my keyboard as much as possible during my work since I find it more efficient then using a mouse/trackpad.
However, if you end up on a product detail page with images in Magento, you can not use your keyboard to navigate to the previous or next page in your browser history.

Previously the fotorama library got triggered for switching to the next or previous image by pressing the left or right arrow key, but when pressing certain modifier keys to go back/forward in the browsers history, fotorama intercepted this and blocked this, which is really annoying.

This PR should fix this.

I researched a little bit what keyboard shortcuts are used in different browsers on different operating systems, and it looks like those are:
- macOS: cmd + left/right arrow key
- linux: alt + left/right arrow key
- windows: alt + left/right arrow key

It looks like all major browsers seem to follow this, but this should be tested by someone having access to all those different variants.
I've only tested on macOS with Google Chrome 69.0.3497.100, Firefox 62.0.2 and Safari 12.0

Also see:
- https://support.google.com/chrome/answer/157179?hl=en
- https://support.mozilla.org/en-US/kb/keyboard-shortcuts-perform-firefox-tasks-quickly#w_navigation
- https://support.microsoft.com/en-us/help/12445/windows-keyboard-shortcuts

The minified fotorama file was manipulated manually because I don't know how the minified file got generated.
I've tested Magento both in development as in production modus (with minify js files enabled) and everything seems to work perfectly fine.

### Fixed Issues (if relevant)
None that I could find.

### Manual testing scenarios
1. Setup a clean Magento installation
2. Add a product in the backend with at least a couple of images (I tested with 3 images)
3. Navigate in your browser to some random page (not the product detail page)
4. Now go to that product detail page
5. Click some link in the footer, just so you have a page in the history of your browser to go forward to from the product detail page
6. Navigate to the previous page (the product detail page) by clicking the back button or keyboard shortcut
7. Test the image gallery, by pressing the left & right arrows, they should work
8. Test the browser navigation by pressing either cmd on macOS or alt on linux/windows + left & right arrow keys

### Contribution checklist
